website/blog/2024-11-27-test-smarter-part-2.md

```diff
@@ -0,0 +1,123 @@
+- Intermediate models may isolate complex operations.
+    - The anomaly tests we list above may suffice here.
+    - You might also consider [unit testing](https://docs.getdbt.com/docs/build/unit-tests) any particularly complex pieces of SQL logic.
+
+### Marts
+
+Marts layer testing will follow the same hygiene-or-anomaly pattern as staging and intermediate. Similar to your intermediate layer, you should focus your testing on net-new columns in your marts layer. This might look like:
+
+- Unit tests: validate especially complex transformation logic. For example:
+    - Calculating dates in a way that feeds into forecasting.
+    - Customer segmentation logic, especially logic that has a lot of CASE-WHEN statements.
+- Primary key tests: focus on where where your mart's granularity has changed from its staging/intermediate inputs.
+    - Similar to the intermediate models above, you may also want to add primary key tests to models whose grain hasn’t changed, but have been enriched with other data. Primary key tests here communicate your intent.
+- Business focused anomaly tests: focus on *new* calculated fields, such as:
+    - Singular tests on high-priority, high-impact tables where you have a specific problem you want forewarning about.
+        - This might be something like fuzzy matching logic to detect when the same person is making multiple emails to extend a free trial beyond its acceptable end date.
+    - A test for calculated numerical fields that shouldn’t vary by more than certain percentage in a week.
+    - A calculated ledger table that follows certain business rules, i.e. today’s running total of spend must always be greater than yesterday’s.
+
+### CI/CD
+
+All of the testing you’ve applied in your different layers is the manual work of constructing your framework. CI/CD is where it gets automated. 
+
+You should run a [slim CI](https://docs.getdbt.com/best-practices/best-practice-workflows#run-only-modified-models-to-test-changes-slim-ci) by default, unless there’s good reason to build everything in your project. 
+
+With CI/CD and your regular production runs, your testing framework can be on autopilot. 😎
+
+If and when you encounter failures, consult your trusty testing framework doc you built in our [earlier post.](https://docs.getdbt.com/blog/test-smarter-not-harder)
+
+### Advanced CI
+
+In the early stages of your smarter testing journey, start with dbt Cloud’s built-in flags for [advanced CI](https://www.notion.so/Test-Smarter-Where-should-tests-go-in-your-pipeline-128bb38ebda7803bb5c2e274c49b2599?pvs=21). In PRs with advanced CI enabled, dbt Cloud will flag what has been modified, added, or removed in the “compare changes” section. These three flags offer confidence and evidence that your changes are what you expect. Then, hand them off for peer review. Advanced CI helps jump start your colleague’s review of your work by bringing all of the implications of the change into one place. 
```

Comment:
This is a notion link - idk where it's meant to go!

Reply:
oh god WHY. fixing!